casual/casual-event/src/test/groovy/se/laz/casual/event/ServiceCallEventImplTest.groovy

```diff
@@ -5,21 +5,83 @@
+   def secondCode = ErrorState.TPENOENT.value
+   def firstOrder = Order.SEQUENTIAL
+   def secondOrder = Order.CONCURRENT
+    
    def 'identity'()
    {
+      given:
+
+      when:
+      ServiceCallEventImpl firstEntry = createEntry([service: firstService, parent: firstParent, pid: firstPid, execution: firstExecution, transactionId: firstTransactionId, start: firstStart, end: firstEnd, pending: firstPending, code: firstCode, order: firstOrder])
+      ServiceCallEventImpl secondEntry = createEntry([service: secondService, parent: secondParent, pid: secondPid, execution: secondExecution, transactionId: secondTransactionId, start: secondStart, end: secondEnd, pending: secondPending, code: secondCode, order: secondOrder])
+      then:
+      firstEntry == firstEntry
+      firstEntry != secondEntry
+
+      firstEntry.getService() == firstService
+      firstEntry.getParent().get() == firstParent
+      firstEntry.getProcessId() == firstPid
+      firstEntry.getExecution() == firstExecution
+      firstEntry.getTransactionId() == firstTransactionId
+      firstEntry.getStart() == firstStart
+      firstEntry.getEnd() == firstEnd
+      firstEntry.getPending() == firstPending
+      firstEntry.getCode() == firstCode
+      firstEntry.getOrder() == firstOrder
+
+      secondEntry.getService() == secondService
+      secondEntry.getParent().get() == secondParent
+      secondEntry.getProcessId() == secondPid
+      secondEntry.getExecution() == secondExecution
+      secondEntry.getTransactionId() == secondTransactionId
+      secondEntry.getStart() == secondStart
+      secondEntry.getEnd() == secondEnd
+      secondEntry.getPending() == secondPending
+      secondEntry.getCode() == secondCode
+      secondEntry.getOrder() == secondOrder
+   }
+
+   def 'hashcode'()
+    {
+      given:
+      ServiceCallEventImpl firstEntry = createEntry([service: firstService, parent: firstParent, pid: firstPid, execution: firstExecution, transactionId: firstTransactionId, start: firstStart, end: firstEnd, pending: firstPending, code: firstCode, order: firstOrder])
+      ServiceCallEventImpl secondEntry = createEntry([service: secondService, parent: secondParent, pid: secondPid, execution: secondExecution, transactionId: secondTransactionId, start: secondStart, end: secondEnd, pending: secondPending, code: secondCode, order: secondOrder])
       when:
-      ServiceCallEventImpl firstEntry = createEntry([service:'service one', parent:'Elvira', pid: 13245, execution: UUID.randomUUID(), transactionId: Mock(Xid), start: 1, end: 42, pending: 0, code: ErrorState.OK.value, order: Order.SEQUENTIAL])
-      ServiceCallEventImpl secondEntry = createEntry([service:'service two', parent:'Elvis', pid: 132456, execution: UUID.randomUUID(), transactionId: Mock(Xid), start: 43, end: 142, pending: 14, code: ErrorState.TPENOENT.value, order: Order.CONCURRENT])
+      Map<ServiceCallEvent, Boolean> map = new HashMap<>()
+      map.put(firstEntry, true)
```

Comment:
Using a map that uses hashCode to hash the keys.
I can change to call hashCode ourself if you want.

Reply:
yes, but it guarentees nothing in this test as the hashcode could return 1 for everything and retrieval would fail.
 The normal test is to ensure hashcode is same for all calls to the same object, so hashcode is not changing without state changing and to ensure that objects that are not equal also do not have a matching hashcode. hence the two tests that i mentioned.